### DISCUSSION TITLE
What's the right way to call down into click.get_current_context ?

### DISCUSSION THREAD

**Discussion by alexrosenfeld10** [fastapi/typer#889] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# this gives me a warning because I've installed typer, not click
from click import get_current_context

def my_thing_that_needs_context():
    ctx = get_current_context()
```

### Description

Is this the right thing to do? Is there something inside typer that exposes the current context functionality, without having to go to click directly myself?

### Operating System

macOS

### Typer Version

0.12.3

### Python Version

3.12.3

**Comment by svlandeg** [comment, marked as answer]
Hi!

> this gives me a warning because I've installed typer, not click

I'm confused why you would get a warning here, as `click` is automatically installed whenever `typer` is installed. So the call to `click.get_current_context()` should just work.

**Reply by alexrosenfeld10** [reply]
it does _work_, but my IDE warns about it. I wonder if this is just a bug in my IDE. If this is the recommended approach that's fine, I'll close this out.
